ui/src/Pages/SearchBar.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from "react";
+import { TextField, IconButton, Button } from "@mui/material";
+import { SearchOutlined } from "@mui/icons-material";
+import Auth from "../utils/Auth";
+import NavBar from "../components/NavBar";
+import Player from "../components/Player";
+import { Playlists } from "../components/Playlists";
+import { Tracks } from "../components/Tracks";
+import { Albums } from "../components/Albums";
+import { Artists } from "../components/Artists";
+import { Artist, Track, Album, Playlist, API } from "../types";
+
+export default function SearchBar() {
+  const [searchInput, setSearchInput] = useState("");
+  const [tracks, setTracks] = useState<Track[]>([]);
+  const [artists, setArtists] = useState<Artist[]>([]);
+  const [albums, setAlbums] = useState<Album[]>([]);
+  const [playlists, setPlaylists] = useState<Playlist[]>([]);
+  const [option, setOption] = useState("");
+  const accessToken = Auth();
+
+  const searchSpotify = () => {
+    const searchParameters = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`,
+      },
+    };
+
+    fetch(
+      `https://api.spotify.com/v1/search?q=${searchInput}&type=artist,album,track,playlist`,
+      searchParameters
+    )
+      .then((response) => response.json())
+      .then((data: API) => {
+        console.log(data);
+        setArtists(data.artists.items);
+        setAlbums(data.albums.items);
+        setTracks(data.tracks.items);
+        setPlaylists(data.playlists.items);
```

Comment:
If you are showing only one set of information at a time, the it is better to fetch only the information which is currently being displayed.